src/workload.rs

```diff
@@ -118,28 +120,32 @@ impl fmt::Display for Upstream {
     }
 }
 
-fn byte_to_ip(b: bytes::Bytes) -> Result<Option<IpAddr>, WorkloadError> {
+fn byte_to_ip(b: &bytes::Bytes) -> Result<Option<IpAddr>, WorkloadError> {
     Ok(match b.len() {
         0 => None,
         4 => {
-            let v: [u8; 4] = (&*b).try_into().expect("size already proven");
-            Some(IpAddr::V4(Ipv4Addr::from(v)))
+            let v: [u8; 4] = b.deref().try_into().expect("size already proven");
+            Some(IpAddr::from(v))
         }
         16 => {
-            let v: [u8; 16] = (&*b).try_into().expect("size already proven");
-            Some(IpAddr::V6(Ipv6Addr::from(v)))
+            let v: [u8; 16] = b.deref().try_into().expect("size already proven");
+            Some(IpAddr::from(v))
+        }
+        n => {
+            println!("invalid ip address length: {n}");
```

Comment:
can we remove print? At the very least we should using `tracing::warning!` probably, but not sure we need it since we already have a ByteAddressParse which should get bubbled up somewhere